cobra/io/sbml.py

```diff
@@ -1487,23 +1516,67 @@ def validate_sbml_model(filename,
         elif sev == libsbml.LIBSBML_SEV_WARNING:
             errors["SBML_WARNING"].append(msg)
 
-    # ensure can be made into model
+    # [2] cobrapy validation (check that SBML can be read into model)
     # all warnings generated while loading will be logged as errors
-    with catch_warnings(record=True) as warning_list:
-        simplefilter("always")
-        try:
-            model = _sbml_to_model(doc)
-        except CobraSBMLError as e:
-            errors["SBML errors"].append(str(e))
-            return None, errors
-        except Exception as e:
-            errors["other"].append(str(e))
-            return None, errors
-
-    errors["warnings"].extend(str(i.message) for i in warning_list)
+    log_stream = StringIO()
+    stream_handler = logging.StreamHandler(log_stream)
+    formatter = logging.Formatter('%(levelname)s:%(message)s')
+    stream_handler.setFormatter(formatter)
+    stream_handler.setLevel(logging.INFO)
+    LOGGER.addHandler(stream_handler)
+    LOGGER.propagate = False
```

Comment:
I wonder if the output should not be visible in parent loggers in addition to being collected from this handler?

Reply:
Yes, I am not sure about this either.

For me it felt cleaner to have the following behavior: 
* If one reads a model with `read_sbml_file` everything is logged, i.e. all warnings and errors. This often results in a lot of warnings in the console (which also takes quit some time in things like notebooks). This is what users normally do. Only if there are are problems calls to `validate_sbml_file` are performed. 
* A call to `validate_sbml_model` is more of a concrete inspection of the model which should provide access to the error and warning information in a simple manner. Users do not necessarily wanting to log all these warnings and errors. The only important information for logging is if there were errors or warnings, all the details are available via the error log.

Reply:
That sounds reasonable to me. Let's leave it as `False`.